src/main/java/com/netflix/imflibrary/st2067_2/AbstractApplicationComposition.java

```diff
@@ -236,6 +226,25 @@ private static Map<UUID, List<IMFBaseResourceType>> getVirtualTrackResourceMap(@
                 if (virtualTrackResourceMap.get(uuid) == null) {
                     virtualTrackResourceMap.put(uuid, new ArrayList<IMFBaseResourceType>());
                 }
+                /*
+                 Ensure that no two resources use the same ID, unless they are the same resource. ST-2067-3:2020, 6.11.1
```

Comment:
Based on this statement the equals and hashcode methods for the resource should include all the elements defined in the resource.

Reply:
I've updated the PR so that the equals and hashcode methods include all elements.

Reply:
I would interpret 6.11.1 as meaning that Resource IDs are unique throughout the _entire composition_:
_"No two Resouces shall have the same ID value unless they are identical."_

It looks like this check is limited to detecting duplicates only _within the same Sequence_.
As a practical matter, that is generally good enough. But I believe there are other possible scenarios that this check would still miss.